homework-submission/react-hyf/react-hyf-client/src/HomeworkWeek3.js

```diff
@@ -0,0 +1,88 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import Name from './Name';
+import Info from './Info';
+import ProfilePicture from './ProfilePicture';
+import Back from './Back';
+
+class HomeworkWeek3 extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      apiData: [],
+      randomPerson: {},
+      picture: '',
+    };
+  }
+
+  componentDidMount = e => {
+    axios
+      .get('https://uinames.com/api/?amount=10')
+      .then(response => {
+        this.setState({ apiData: response.data });
+      })
+      .catch(error => {
+        console.log(error);
+      })
+      .then(response => {
+        this.randomPerson();
+      })
+      .then(response => {
+        this.getPicture();
+      });
+  };
+
+  getPicture = e => {
```

Comment:
Why are you separately getting the user picture (and recursively calling the function to do so)? Doesn't the initial api call response include the picture?

Reply:
no actually this is what i get by this api
[{"name":"Gary","surname":"Greene","gender":"male","region":"United States"},{"name":"Pécsi","surname":"Pál","gender":"male","region":"Hungary"},{"name":"Jesse","surname":"Harwood","gender":"male","region":"New Zealand"},{"name":"Vương","surname":"Cúc","gender":"female","region":"Vietnam"},{"name":"Zoe","surname":"McKenzie","gender":"female","region":"New Zealand"},{"name":"Miriama","surname":"Kolník","gender":"female","region":"Slovakia"},{"name":"Stefan","surname":"Huber","gender":"male","region":"Austria"},{"name":"Loane","surname":"Charpentier","gender":"female","region":"France"},{"name":"Jamie","surname":"Wolf","gender":"male","region":"Germany"},{"name":"Blažej","surname":"Malíšek","gender":"male","region":"Slovakia"}]

Reply:
i took me a lot of time to find a free api that gives random photo's and then another http request and i had to do spend a lot of time to figure out how to match the photo gender with the gender from the other request

Reply:
I understand now, in that case you solution -although not so efficient- is pretty smart. Finding a separate API and working with that to get the pictures must have been challenging. Still I'd like to ask you to carefully read the documentation of https://uinames.com (it's behind the menu button in the top right). It is possible to get a picture from that API.

Since you are probably already busy with the project, it's also fine if you explain here how you could use uinames.com to get the photo data. Or feel free to code it of course, if you want to give it a try.

Good luck!

Reply:
I have updated the homework
I created a new component named homeworkupdate where i applied many changes 
hope you like it
let me know if it's ok so